cr-usecases/src/main/java/com/thefuntasty/mvvm/crusecases/CoroutineScopeOwner.kt

```diff
@@ -193,6 +227,45 @@ interface CoroutineScopeOwner {
             .launchIn(coroutineScope)
     }
 
+    /**
+     * Launch suspend [block] in [coroutineScope]. Encapsulates this call with try catch block and when an exception is thrown
+     * then it is logged in [logUnhandledException] and handled by [defaultErrorHandler].
+     *
+     * If exception is [CancellationException] then [defaultErrorHandler] is not called and [logUnhandledException] is called
+     * only if the root cause of this exception is not [CancellationException] (e.g. when [Result.getOrCancel] is used).
+     */
+    fun launchWithHandler(block: suspend CoroutineScope.() -> Unit) {
+        coroutineScope.launch {
+            try {
+                block()
+            } catch (exception: CancellationException) {
+                val rootCause = exception.rootCause
+                if (rootCause != null && rootCause !is CancellationException) {
+                    logUnhandledException(exception)
+                }
+            } catch (exception: Throwable) {
+                logUnhandledException(exception)
```

Comment:
Can you create the same Global error handle (or maybe use it here too) as you've created for Rx version? 

https://github.com/futuredapp/mvvm-android/blob/e4ebc2f6e4960a2c4ffbb7487b4643f5f12f67aa/rx-usecases/src/main/java/com/thefuntasty/mvvm/rxusecases/UseCaseErrorHandler.kt

Reply:
I was thinking that I will leave it up to the developer to do it by himself in ViewModel, but Global one will be better 👍

Reply:
Yep, you as a developer don't want to handle this is every single ViewModel.